File: app/src/main/java/fr/erictruong/material/training/ui/components/cards/CardsFragment.java
package fr.erictruong.material.training.ui.components.cards;

import android.os.Bundle;
import android.view.View;

import java.util.ArrayList;
import java.util.List;

import fr.erictruong.android.cards.MaterialCardItem;
import fr.erictruong.android.cards.item.DisplayBodyItem;
import fr.erictruong.android.cards.item.HeadlineBodyItem;
import fr.erictruong.material.training.R;
import fr.erictruong.material.training.ui.core.BaseCardFragment;

import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_AVATAR_TITLE_SUBTITLE_IMAGE_BODY1_SIX_BUTTON;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_AVATAR_TITLE_SUBTITLE_IMAGE_BODY1_THREE_BUTTON;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_BACKGROUND_HEADLINE_SUBTITLE_BODY1_THREE_BUTTON;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_DISPLAY1_PRIMARY_BODY2;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_HEADLINE_SUBTITLE_BODY1_THREE_BUTTON;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_HEADLINE_SUBTITLE_IMAGE_THREE_BUTTON;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_HEADLINE_SUBTITLE_IMAGE_THREE_BUTTON_LARGE;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_HEADLINE_SUBTITLE_IMAGE_THREE_BUTTON_SMALL;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_IMAGE_BODY1;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_IMAGE_HEADLINE_SUBTITLE_BODY1_THREE_BUTTON_EXPAND;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_IMAGE_HEADLINE_THREE_ICON;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_IMAGE_THREE_ICON;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_IMAGE_THREE_ICON_VERTICAL;

public class CardsFragment extends BaseCardFragment {

    public static CardsFragment newInstance() {
        Bundle args = new Bundle();
        CardsFragment fragment = new CardsFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        List<MaterialCardItem> cards = new ArrayList<>();
        cards.add(new DisplayBodyItem.Builder()
                .viewType(VIEW_TYPE_DISPLAY1_PRIMARY_BODY2)
                .display(getString(R.string.cards))
                .body(getString(R.string.cards_txt))
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Media area - Supporting text")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_IMAGE_BODY1);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Avatar, Title, and Subtitle area - Media area - Supporting text - Actions")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_AVATAR_TITLE_SUBTITLE_IMAGE_BODY1_THREE_BUTTON);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Avatar, Title, and Subtitle area - Media area - Actions")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_AVATAR_TITLE_SUBTITLE_IMAGE_BODY1_SIX_BUTTON);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Media area - Primary text - Subtext - Actions - Expanded supporting text")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_IMAGE_HEADLINE_SUBTITLE_BODY1_THREE_BUTTON_EXPAND);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Primary text - Subtext - Supporting text - Actions")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_HEADLINE_SUBTITLE_BODY1_THREE_BUTTON   );
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Media area - Actions")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_IMAGE_THREE_ICON);
                    }
                })
                .button2("Alt", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_IMAGE_THREE_ICON_VERTICAL);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Media area - Primary text - Subtext - Actions")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_BACKGROUND_HEADLINE_SUBTITLE_BODY1_THREE_BUTTON);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Media area - Primary text - Actions")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_IMAGE_HEADLINE_THREE_ICON);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Media area - Primary text - Subtext - Actions")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_HEADLINE_SUBTITLE_IMAGE_THREE_BUTTON_SMALL);
                    }
                })
                .button2("Alt #2", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_HEADLINE_SUBTITLE_IMAGE_THREE_BUTTON);
                    }
                })
                .button3("Alt #3", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleCardActivity.start(getContext(), VIEW_TYPE_HEADLINE_SUBTITLE_IMAGE_THREE_BUTTON_LARGE);
                    }
                })
                .build());

        getAdapter().swapData(cards);
    }
}

File: app/src/main/java/fr/erictruong/material/training/ui/components/cards/CardsFragment.kt
package fr.erictruong.material.training.ui.components.cards

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.CardView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import fr.erictruong.material.training.R
import org.jetbrains.anko.*
import org.jetbrains.anko.support.v4.nestedScrollView

class CardsFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return CardsFragmentUI().createView(AnkoContext.create(context, this))
    }
}

class CardsFragmentUI : AnkoComponent<Fragment> {

    override fun createView(ui: AnkoContext<Fragment>) = with(ui) {
        nestedScrollView {
            verticalLayout {
                padding = dimen(R.dimen.padding_medium)
                bottomPadding += dimen(R.dimen.action_bar_size)
                clipToPadding = false

                textView("Media area - Supporting text")
                include<CardView>(R.layout.card_media_body1)
                textView("Avatar, Title, and Subtitle area - Media area - Supporting text - Actions")
                include<CardView>(R.layout.card_avatar_subheader_media_body1_actions)
                textView("Avatar, Title, and Subtitle area - Media area - Actions")
                include<CardView>(R.layout.card_avatar_subheader_media_body1_vertical_actions)
                textView("Media area - Primary text - Subtext - Actions - Expanded supporting text")
                include<CardView>(R.layout.card_media_headline_subtext_actions_expanded_body1)
                textView("Primary text - Subtext - Supporting text - Actions")
                include<CardView>(R.layout.card_headline_subtext_body1_actions)
                textView("Media area - Actions #1")
                include<CardView>(R.layout.card_media_actions_alt1)
                textView("Media area - Actions #2")
                include<CardView>(R.layout.card_media_actions_alt2)
                textView("Media area - Primary text - Actions")
                linearLayout {
                    clipToPadding = false

                    include<CardView>(R.layout.card_media_headline_actions).lparams {
                        width = 0
                        weight = 1f
                    }
                    space {}.lparams(width = dip(8), height = dip(8))
                    include<CardView>(R.layout.card_media_headline_actions).lparams {
                        width = 0
                        weight = 1f
                    }
                }
                textView("Media area - Primary text - Subtext - Actions #1")
                include<CardView>(R.layout.card_media_headline_subtext_actions_alt1)
                textView("Media area - Primary text - Subtext - Actions #2")
                include<CardView>(R.layout.card_media_headline_subtext_actions_alt2)
                textView("Media area - Primary text - Subtext - Actions #3")
                include<CardView>(R.layout.card_media_headline_subtext_actions_alt3)
                textView("Media area - Primary text - Subtext - Actions #4")
                include<CardView>(R.layout.card_media_headline_subtext_actions_alt4)
//                button("Media area - Supporting text") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Avatar, Title, and Subtitle area - Media area - Supporting text - Actions") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Avatar, Title, and Subtitle area - Media area - Actions") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Media area - Primary text - Subtext - Actions - Expanded supporting text") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Primary text - Subtext - Supporting text - Actions") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Media area - Actions #1") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Media area - Actions #2") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Media area - Primary text - Subtext - Actions") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Media area - Primary text - Actions") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Media area - Primary text - Subtext - Actions #1") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Media area - Primary text - Subtext - Actions #2") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
//                button("Media area - Primary text - Subtext - Actions #3") { onClick { /*ctx.startActivity<SampleCardActivity>()*/ } }.enabled = false
            }
        }
    }
}
